src/rscminus/game/entity/Player.java

```diff
@@ -656,8 +664,21 @@ public void processIncomingPackets() {
                 m_actionSlot.setAction(ActionSlot.ACTION_INVENTORY_DROP);
                 m_actionSlot.setInventorySlot(m_packetStream.readUnsignedShort());
                 break;
+            case OPCODE_PUBLIC_CHAT:
+                int messageLength = m_packetStream.readVariableSize();
+                if (messageLength <= 0 || messageLength > 80)
+                    break;
+                String message = StringEncryption.decipher(m_packetStream, messageLength);
+                //TODO: Profanity filter
```

Comment:
Besides just Profanity filter, you also need to handle not being able to Shout & forced capitalization of the beginning of sentences.

Reply:
The chat censor from 204 was implemented and a capitalization filter was added based on findings from your message dump. The cap filter is tested by an included unit test, "chat"